backend/src/services/project/project-service.ts

```diff
@@ -360,7 +364,7 @@ export const projectServiceFactory = ({
 
       return {
         ...project,
-        environments: envs,
+        environments: envs || [],
```

Comment:
I would initialize `envs` as an array to begin with. Then you won't have to do this `or` statement